barretenberg/cpp/src/barretenberg/vm2/tracegen/precomputed_trace.cpp

```diff
@@ -215,6 +216,82 @@ void PrecomputedTraceBuilder::process_wire_instruction_spec(TraceContainer& trac
     }
 }
 
+void PrecomputedTraceBuilder::process_exec_instruction_spec(TraceContainer& trace)
+{
+    using C = Column;
+    // todo(ilyas): change this to constexpr when the final exec instruction spec is available
+    uint32_t num_opcodes = static_cast<uint32_t>(EXEC_INSTRUCTION_SPEC.size());
+    trace.reserve_column(C::precomputed_exec_opcode_value, num_opcodes);
+    // Gas Costs
+    trace.reserve_column(C::precomputed_exec_opcode_base_l2_gas, num_opcodes);
+    trace.reserve_column(C::precomputed_exec_opcode_base_da_gas, num_opcodes);
+    trace.reserve_column(C::precomputed_exec_opcode_dynamic_l2_gas, num_opcodes);
+    trace.reserve_column(C::precomputed_exec_opcode_dynamic_da_gas, num_opcodes);
+    // Memory Access for Registers
+    trace.reserve_column(C::precomputed_mem_op_reg1, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_1, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg2, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_2, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg3, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_3, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg4, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_4, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg5, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_5, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg6, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_6, num_opcodes);
+    trace.reserve_column(C::precomputed_mem_op_reg7, num_opcodes);
+    trace.reserve_column(C::precomputed_rw_7, num_opcodes);
+    // Gadget/Subtrace Selectors
+    trace.reserve_column(C::precomputed_op_alu_sel, num_opcodes);
+    trace.reserve_column(C::precomputed_op_bitwise_sel, num_opcodes);
+    trace.reserve_column(C::precomputed_op_poseidon_perm_sel, num_opcodes);
+    trace.reserve_column(C::precomputed_op_to_radix_sel, num_opcodes);
+    trace.reserve_column(C::precomputed_subtrace_operation_id, num_opcodes);
```

Comment:
I think in practice this doesn't change much for such a short trace. You can keep it or remove it.

Reply:
ah good to know - ill remove to keep it more readable then